src/events.ts

```diff
@@ -1,41 +1,50 @@
-enum messages {
+enum partnerEvents {
   // It's really unfortunate that we call this event "accountProvisioned",
   // when what we mean is "accountProvisionInitiated". But that's the current
   // name that is being used by the dreams backend
-  accountProvisioned = 'accountProvisioned',
+  accountProvisionInitiated = 'accountProvisioned',
+  investmentAccountProvisionInitiated = 'investmentAccountProvisionInitiated',
   updateToken = 'updateToken',
   navigateTo = 'navigateTo'
 }
 
 type Message = {
-  message: {
-    requestId: string;
-    idToken?: string;
-  };
+  requestId: string;
 }
 
-type ShareMessage = {
-  message: {
-    url: string;
-    text: string;
-    title: string;
-  };
+type ShareMessage = Message & {
+  url: string;
+  text: string;
+  title: string;
 }
 
-type MessageEvent = Message & {
-  event: messages;
+type TokenDidExpireMessage = Message & {
+  idToken: string;
 }
 
-type IdTokenDidExpireEvent =  Message & {
+type InvestmentAccountProvisionMessage = Message & {
```

Comment:
To be consistent with other naming: => `InvestmentAccountProvisionRequestedMessage` ?